src/main/java/de/adesso/anki/battle/GameEngine.java

```diff
@@ -145,34 +184,83 @@ private void updateSimulation(long deltaNanos) {
            body.updatePosition(deltaNanos);
         }
     }
+	
+	
+    private void collectOrphanedWeapons() {
+    	Iterator<Body> it = world.getBodiesModifiable().iterator();
+        while(it.hasNext()){
+        	Body weapon = it.next();
+        	if (weapon instanceof Rocket && ((Rocket) weapon).shouldExplode()) {
+        		it.remove();
+        		System.out.println("Delete orphaned rocket");
+        	}
+        }
+	}
 
-    
+	
+	//merge though filter predicate with method above
+	private void collisionHandling() {
+		Iterator<Body> it = world.getBodiesModifiable().iterator();
+        while(it.hasNext()){
+        	Body weapon = it.next();
+        	if (checkCollision(weapon)) {
+        		it.remove();
+        	}
+        }
+    }
 
-    private void collisionHandling() {
-    	// TODO 
+	private boolean checkCollision(Body weapon) {
+		if ( weapon instanceof Vehicle) {
+			return false;
+		}
+		// merge into new superclass weapon ? 
+    	if (weapon instanceof Rocket && !((Rocket) weapon).isActive()) {
+    		return false;
+    	}
+    	if (weapon instanceof Mine && !((Mine) weapon).isActive()) {
+    		return false;
+    	}
+    	
+    	List<Vehicle> vehicles = world.getVehicles();
+		Position pos1 = weapon.getPosition();
+		boolean succesfulHit = false;
+		//TODO find damage values for weapon types
+		int damage = ((weapon instanceof Rocket ) ? 10 : 20);
+    	for (Vehicle vehicle : vehicles) {
+			Position pos2 = vehicle.getPosition();
+			double distance = pos1.distance(pos2);
+			//TODO find distance value that indicates a collision

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)